### PR TITLE
Database snapshot support

### PR DESCRIPTION
**Pull request by murrant**
Improve the speed of initial database migration and tests

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Console/MigrateCommand.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    protected function prepareDatabase()
+    {
+        $database = $this->option('database');
+        $this->migrator->setConnection($database);
+
+        // import snapshot if this is a new DB and it is available
+        if (!$this->option('no-snapshot')
+            && !$this->migrator->repositoryExists()
+            && is_file(\config('snipe.snapshot-location'))
```
Comment (Jellyfrog) [inline app/Console/MigrateCommand.php]: wrong path

File: app/Console/MigrateCommand.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        ) {
+            $this->output->write('Importing DB snapshot... ');
+            try {
+                $res = $this->laravel['db']->connection($database)
+                    ->getPdo()->exec(file_get_contents(\config('snipe.snapshot-location')));
```
Comment (Jellyfrog) [inline app/Console/MigrateCommand.php]: wrong path

**Comment by Jellyfrog** [comment]
Spontaneously it seems annoying to maintain the `misc/snapshot.sql`  
I did a quick and dirty test on a virtual machine on my laptop:
- `time php artisan migrate` = 22s
- `time mysql librenms < librenms.sql` = 9s

With `innodb_flush_log_at_trx_commit = 0` (can be used in test) in my.cnf:
- `time php artisan migrate` = 13s
- `time mysql librenms < librenms.sql` = 8s

File: app/Console/MigrateCommand.php (hunk 2 above)
Comment (murrant) [reply app/Console/MigrateCommand.php]: opsie :)  nice catch

**Comment by Jellyfrog** [comment]
Close in favor of https://github.com/laravel/framework/pull/32275?

**Comment by murrant** [comment]
Yep.

**Pull request closed without merging**